client/client-multi/client-cpp/impl/bootstrap/BootstrapManager.cpp

```diff
@@ -56,55 +56,69 @@ BootstrapManager::OperationsServers BootstrapManager::getOPSByAccessPointId(std:
+                                  });
+            break;
+        }
+        case FailoverStrategyAction::USE_NEXT_OPERATIONS_SERVER:
+        {
+            OperationsServers::iterator nextOperationIterator = (lastServerIt->second) + 1;
+            if (nextOperationIterator != serverIt->second.end()) {
+                KAA_LOG_INFO(boost::format("New Operations server [%1%] will be used for %2%")
+                                           % (*nextOperationIterator)->getAccessPointId()
+                                           % LoggingUtils::toString(protocolId));
 
-            lastOperationsServers_[protocolId] = nextOperationIterator;
+                lastOperationsServers_[protocolId] = nextOperationIterator;
 
-            if (channelManager_ != nullptr) {
-                channelManager_->onTransportConnectionInfoUpdated(*(nextOperationIterator));
+                channelManager_->onTransportConnectionInfoUpdated(*nextOperationIterator);
             } else {
-                KAA_LOG_ERROR("Can not process server change. Channel manager was not specified");
-            }
-        } else {
-            KAA_LOG_WARN(boost::format("Failed to find server for channel %1%.")
-                         % LoggingUtils::TransportProtocolIdToString(protocolId));
-
-            FailoverStrategyDecision decision = failoverStrategy_->onFailover(reason);
-            switch (decision.getAction()) {
-                case FailoverStrategyAction::NOOP:
-                    KAA_LOG_WARN("No operation is performed according to failover strategy decision.");
-                    break;
-                case FailoverStrategyAction::RETRY:
-                {
-                    std::size_t period = decision.getRetryPeriod();
-                    KAA_LOG_WARN(boost::format("Attempt to receive operations server list will be made in %1% secs "
-                                               "according to failover strategy decision.") % period);
-                    retryTimer_.stop();
-                    retryTimer_.start(period, [&] { receiveOperationsServerList(); });
-                    break;
-                }
-                case FailoverStrategyAction::STOP_CLIENT:
-                    KAA_LOG_WARN("Stopping client according to failover strategy decision!");
-                    client_->stop();
-                    break;
-                default:
-                    break;
+                KAA_LOG_WARN(boost::format("No Operations servers are accessible for %1%.")
+                                                         % LoggingUtils::toString(protocolId));
+
+                onCurrentBootstrapServerFailed(KaaFailoverReason::ALL_OPERATIONS_SERVERS_NA);
             }
+            break;
         }
-    } else {
-        throw KaaException("There are no available servers at the time");
+        case FailoverStrategyAction::USE_NEXT_BOOTSTRAP_SERVER:
+            onCurrentBootstrapServerFailed(KaaFailoverReason::ALL_OPERATIONS_SERVERS_NA);
```

Comment:
A bit confusing to me, since according to this method name it is an operation server that fails.